common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -30,9 +30,9 @@
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 
-public class DefaultPromise<V> implements Promise<V> {
-    private static final InternalLogger logger = InternalLoggerFactory.getInstance(DefaultPromise.class);
-    private static final InternalLogger rejectedExecutionLogger =
+public class DefaultPromise<V> implements Promise<V>, FutureContextListener<PromiseOperator<V>, Object> {
```

Comment:
I am not thrilled by adding this detail about how we implement operators on a public class.

Perhaps extend `DefaultPromise` in a package-private class?